### DISCUSSION TITLE
More query in where of update function

### DISCUSSION THREAD

**Discussion by IRediTOTO** [prisma/prisma#14444] [Q&A] [answered]
Hello, I am trying to update a row like this
```
prm.products.update(
where:{ 
id:"abc",
user:{id: 5}  // I want check user is owner here 
}
data:{...}
)
```
But I want check current user is owner of this product. How can I do that? I don't want query product again to check owner.
Here is my schema
```
model Products {
  seller        String
  User          User           @relation(fields: [seller], references: [id])
}

model User {
....
  Products                          Products[]

}

```

**Comment by nurul3101** [comment, marked as answer]
Hey @IRediTOTO 👋 

If I understand correctly you only want product owner to be able to update the product, could you store an ownerId field in the product model and while updating the product only get the products which match with the ownerId?
Like this
```typescript
prm.products.update(
where:{ 
id:"abc",
ownerId: "5" 
}
data:{...}
)
```

**Reply by IRediTOTO** [reply]
But the `ownerId` is not unique, it isn't showed in `where`. I can use `updateMany` work like that, but in `data` i can't set data for relation field like tag/category

**Reply by SamuelMS** [reply]
@nurul3101 That of course works, but copying around / denormalizing a field like `ownerId` can lead to developer error / data synchronicity issues. I feel like it's pretty natural to want to run updates on related data – where the owner is the same as the person is making a query, in this case – and avoid having to explicitly pull down the record ahead of time with an extra call.

Is it a heavy lift to switch `.update`'s `where` clause from `findUnique` to `findFirst`? If the update fails to find a record, that's completely OK / very understandable behavior. (Another workaround is to do this via raw SQL, but that mostly defeats the point of using prisma in the first place.)

This would be a _major_ value add for my team.

**Reply by pantharshit00** [reply]
Hello,

This is now possible with https://github.com/prisma/prisma/releases/tag/4.5.0 release with `extendedWhereUnique` preview flag. I am marking this discussion as solved.

**Reply by SamuelMS** [reply]
@pantharshit00 Amazing, thank you. This is a huge quality of life improvement.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
